### PR TITLE
Fix followed_by due to API change

### LINKED ISSUE #1194: Profiler partially failing
Timezone and mutual following are currently stuck on infinite loading. Part of the problem is that there is one API call not done via HTTPS, which triggers a MixedContent error. If that bug is fixed, the attempt to parse the received data into JSON (https://github.com/new-xkit/XKit/blob/master/Extensions/profiler.js#L440) fails, and stops the extension altogether. Additionally, a 404 error is thrown while trying to access the `followed_by` SVC call.

High priority since it breaks a significant part of the extension.

### PR DESCRIPTION
**Pull request by nightpool**
Fix followed_by due to the new API change. adds a new interface function to encapsulate this call.

Fixes glowing follow and profiler.

Closes #1194

### CODE REVIEW

File: Extensions/xkit_patches.js [hunk 1]
```diff
@@ -2221,6 +2221,23 @@ XKit.tools.getParameterByName = function(name){
 				XKit.tools.add_function(function(){
 					Tumblr.Events.trigger("peepr-open-request", add_tag);
 				}, true, payload);
+			},
+
+			/**
+			 * Determines whether a user is following the given blog.
+			 * The logged-in user must be a member of the given blog to determine this.
```
Comment (blackjackkent) [inline Extensions/xkit_patches.js]: What's the behavior if the logged-in user ISN'T a member of the blog? Do we need an error handler here?

**Comment by blackjackkent** [comment]
Added one quick comment/question. Otherwise looks good.

**Comment by nightpool** [comment]
Okay, I wasn't able to clean up the error handling, because jQuery promises implement error handling very weirdly. Everything else should be g2g though. I removed the explicit JSON.parse by changing the type to "json" in the jquery call.

File: Extensions/profiler.js [hunk 2]
```diff
@@ -334,22 +334,11 @@ XKit.extensions.profiler = new Object({
...
-			try {
-				msg = JSON.parse(msg);
-				if (msg.response.is_friend === 1) {
-					$("#xkit-profiler-is-following").removeClass("loading-up").html("Yes");
-				} else {
-					$("#xkit-profiler-is-following").removeClass("loading-up").html("No");
-				}
-			} catch(e){
-				console.log("Profiler: " + e.message);
+		XKit.interface.is_following(username, blog_id).then(function(follow) {
```
Comment (blackjackkent) [inline Extensions/profiler.js]: `username` is undefined here. Should be `user_url` I think?

Comment (nightpool) [inline Extensions/profiler.js]: odd. will fix. swore I tested this....

**Comment by blackjackkent** [comment]
LGTM

@homu r+

**Pull request merged**